### ISSUE TITLE
Automatic selection of segmentation node at loading

### ISSUE THREAD

**Issue by Punzo** [OHIF/Viewers#2332]
The viewer should automatically load SEG in the case when we have a single imaging series and a single SEG. If we have multiple SEGs, and multiple imaging series, would it make sense to automatically load the first SEG.

Currently there are two bugs:

1) none segmentation is selected if we open a study by specifying the series uid (i.e. single imaging series and a single SEG). 
https://dev-viewer.canceridc.dev/viewer/1.3.6.1.4.1.14519.5.2.1.6279.6001.224985459390356936417021464571?seriesInstanceUID=1.2.276.0.7230010.3.1.3.0.57823.1553343864.578877,1.3.6.1.4.1.14519.5.2.1.6279.6001.273525289046256012743471155680   

2) a "random" seg is selected if in the link there is only the study id (i.e. multiple SEGs / multiple imaging series).
https://dev-viewer.canceridc.dev/viewer/1.3.6.1.4.1.14519.5.2.1.6279.6001.224985459390356936417021464571
In this case the **Segmentation of Nodule 3** is selected, while we should select **Segmentation of Nodule 1** (first in the list in the left panel). It looks like **Segmentation of Nodule 3** is loaded as first and assigned as the selected seg. We should assing the selected segmentation once the loading is finished and select the first seg as shown in the left panel of the UI.

**Comment by Punzo** [comment]
> 2. a "random" seg is selected if in the link there is only the study id (i.e. multiple SEGs / multiple imaging series).
>    https://dev-viewer.canceridc.dev/viewer/1.3.6.1.4.1.14519.5.2.1.6279.6001.224985459390356936417021464571
>    In this case the **Segmentation of Nodule 3** is selected, while we should select **Segmentation of Nodule 1** (first in the list in the left panel). It looks like **Segmentation of Nodule 3** is loaded as first and assigned as the selected seg. We should assing the selected segmentation once the loading is finished and select the first seg as shown in the left panel of the UI.

@fedorov regarding point (2). OHIF loads at startup automatically the most recent (in date) SEG for the image series currently shown. Do we want to change this logic? and automatically load the first SEG? or loading the most recent one is fine?

**Comment by Punzo** [comment]
> I think picking the most recent is a good behavior. We also do this for measurement SRs based on the idea that represent multiple versions. For the SEG maybe the series list should also be sorted in reverse time order. If we don't think this covers all cases then we should think about how to encode extra information about sorting and expose options to the users.

ok then I will leave as it is for point (2). 

Yes ordering the SEG in reverse time order makes sense (both in the combobox list in the segmentation panel and the SEG thumbnails in the left panel). Would it make sense to track this in another issue?

**Comment by pieper** [comment]
Let's see if Andrey has other ideas, but I think it's logically okay to keep it as part of this issue.

**Comment by Punzo** [comment]
> The issue I have is that the attributes used to sort are not accessible to the user. I think if we sort by date/time in the combobox, we should add time to the list below. What do you think? Also, there are many different dates/times in DICOM - which one is going to be used?

I agree in adding the time under the name of the items in the combobox list. the time used in OIHF is the attributes:
A) Series Date, attribute (0008,0021)
B) Series Time, attribute (0008,0031)

ok then :
1) I will add the time to the date in the combobox items
2) I will order the combobox items in reverse time order


> For sorting in the left panel, same argument applies - if they are sorted, I think the sort attribute should be displayed next to the thumbnail. I think for the left panel, sorting by SeriesNumber should remain the sort criteria, at least until date/time is possible to display in the left panel. Do you agree?

ok then for the moment I will leave the left panel as it is.

**Comment by fedorov** [comment]
Does this make sense to you? @pieper what do you think? I definitely don't want you to do this just because this is what I said!

**Comment by Punzo** [comment]
yes for me makes sense!

**Comment by Punzo** [comment]
![image](https://user-images.githubusercontent.com/7985338/113006995-1df9a300-9176-11eb-8c3b-48f5ebd388ef.png)

**Comment by fedorov** [comment]
Looks good to me! Indeed we need to show seconds to be able to sort! :-)

**Comment by Punzo** [comment]
PR in https://github.com/OHIF/Viewers/pull/2341

**Comment by fedorov** [comment]
Works in test tier. Great!

**Comment by fedorov** [comment]
This seems to be broken - reopening.

**Comment by fedorov** [comment]
https://dev-viewer.canceridc.dev/viewer/1.3.6.1.4.1.14519.5.2.1.6279.6001.224985459390356936417021464571

**Comment by Punzo** [comment]
closing again, since no regression has been found. The current logic follow the issue description (https://github.com/OHIF/Viewers/issues/2332#issue-841182514): i.e. at first loading of a study, in the case of multiple segmentations, we load only the first segmentation relative to the active source series (segmentations are ordered by time in the combobox list, and 'first' is the most recent SEG in time). This 'first' segmentation is therefore loaded, set as active in the combobox and all the segments of that segmentation get visualized over the source image (see attached video). The others are not loaded automatically, and they are loaded and visualized (either as active or non active) only when the user select them in the combobox or click and drag the SEG thumbnail in the viewport.

See video:
https://user-images.githubusercontent.com/7985338/151055562-a1732eea-faa8-4510-8b22-681d296a59c5.mp4
